### PR TITLE
gh-98230: Add the daemon keyword argument to the Timer constructor

### PR DESCRIPTION
**Pull request by Gouvernathor**
Very straightforward implementation.
Doc change will be made if and when the idea is approved.

* Issue: gh-98230

### CODE REVIEW

**Review by sobolevn — COMMENTED** [review]
Hi, thanks for the PR!

When implementing a new feature, we 100% need to write unit tests for it. It makes chances of being accepted a lot higher :)

Also, don't forget to document the argument you are proposing. 
You would also need to add NEWS. And maybe even adding "What's new?" entry.

File: Misc/NEWS.d/next/Library/2022-10-13-13-32-05.gh-issue-98230.oTb1Az.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+Add a daemon keyword-only argument to the :func:`threading.Timer` constructor, to pass it to the inherited :func:`Thread` constructor.
```
Comment (sobolevn) [inline Misc/NEWS.d/next/Library/2022-10-13-13-32-05.gh-issue-98230.oTb1Az.rst]:
```suggestion
Add a daemon keyword-only argument to the :class:`threading.Timer` constructor, to pass it to the inherited :class:`threading.Thread` constructor.
```

**Comment by Gouvernathor** [comment]
I copied the equivalent test for Thread, it looked simple enough.

**Comment by erlend-aasland** [comment]
Closing, since the linked issue is rejected.

**Pull request closed without merging**